### ISSUE TITLE
Cancelling an observation quickly after starting it results in a crash

### ISSUE THREAD

**Issue by layoutSubviews** [groue/GRDB.swift#1195]
### What did you do?

Our application, in some circumstances, starts DB observations and quickly cancels them as the parameters change.

### What did you expect to happen?

GRDB should tear down these short-lived observations successfully.

### What happened instead?

GRDB frequently crashes at `ValueWriteOnlyObserver.swift:269` which is `fatalError("can't start a cancelled or failed observation")`.

### Environment

**GRDB flavor(s):** Vanilla
**GRDB version:**  v5.22.1
**Installation method:** SPM
**Xcode version:** 13.3
**Swift version:** 5.6
**Platform(s) running GRDB:** iOS
**macOS version running Xcode:** N/A

### Demo Project

```
        var configuration = Configuration()
        configuration.targetQueue = DispatchQueue(label: "crash.test", qos: .userInitiated)

        let database = DatabaseQueue(configuration: configuration)

        var migrator = DatabaseMigrator()
        migrator.registerMigration("v1") { (db) in
            try db.create(table: Record.databaseTableName) { (t) in
                t.autoIncrementedPrimaryKey("id")
            }
        }

        try! migrator.migrate(database)

        let observation = ValueObservation.tracking { (db) in
            try Record.fetchCount(db)
        }

        var cancellable: AnyCancellable?
        cancellable = observation.publisher(in: database, scheduling: .async(onQueue: DispatchQueue.global()))
            .assertNoFailure()
            .sink { value in
                print(value)
            }
        cancellable?.cancel()
```

**Comment by groue** [comment]
Hello @layoutSubviews,

Thank you very much for reporting this regression in 5.22.0, and finding a fix in #1196 - I'll join you there.

Other users should downgrade to 5.21 in order to avoid this issue, until the fix is shipped!

**Comment by groue** [comment]
Fixed in v5.22.2
